network/src/network.rs

```diff
@@ -1104,3 +1130,17 @@ impl Drop for NetworkController {
         self.stop.try_send();
     }
 }
+
+// Send a optional message before disconnect a peer
+pub(crate) fn disconnect(
```

Comment:
Suggest to move this fn to `ServiceControl` and rename it to `disconnect_with_message(&self, peer_index: SessionId, message: &str)`

Reply:
For up level (ckb) maintainability `ServiceControl::disconnect_with_message`. Based on that idea, there are to solutions:
1. Setup a p2p internal protocol for receiving the message (Which is ugly)
2. p2p need a new basic mechanism to receive the message (In current case, I don't it's worth it)

How about this compromised way? :
1. `DisconnectMessageProtocol` still required
2. `ServiceControl::disconnect_with_message(&self, peer_index: SessionId, proto_id: ProtocolId, message: &str)`